awx_collection/plugins/modules/tower_project.py

```diff
@@ -234,21 +233,22 @@ def main():
     wait = module.params.get('wait')
 
     # Attempt to look up the related items the user specified (these will fail the module if not found)
-    org_id = module.resolve_name_to_id('organizations', organization)
-    if credential is not None:
-        credential = module.resolve_name_to_id('credentials', credential)
+    lookup_data = {}
+    org_id = None
+    if organization:
+        org_id = module.resolve_name_to_id('organizations', organization)
+        lookup_data['organization'] = org_id
 
     # Attempt to look up project based on the provided name and org ID
-    project = module.get_one('projects', name_or_id=name, **{
-        'data': {
-            'organization': org_id
-        }
-    })
+    project = module.get_one('projects', name_or_id=name, data=lookup_data)
 
     if state == 'absent':
         # If the state was absent we can let the module delete it if needed, the module will handle exiting from this
         module.delete_if_needed(project)
 
+    if credential is not None:
+        credential = module.resolve_name_to_id('credentials', credential)
```

Comment:
I kind of just made the change suggested in https://github.com/ansible/awx/pull/7925